evalml/tests/component_tests/test_components.py

```diff
@@ -1628,3 +1634,15 @@ def test_estimator_fit_respects_custom_indices(
     estimator.fit(X, y)
     pd.testing.assert_index_equal(X.index, X_original_index)
     pd.testing.assert_index_equal(y.index, y_original_index)
+
+
+def test_component_parameters_supported_by_list_API():
```

Comment:
Testing private attribute... but I think it could be good to have this so that we're aware each time a new component is added :)

Reply:
Sounds good to me!

Reply:
Would we want to just smoke test the component to make sure that those that support the list API do and those that don't, do not?